src/datanode/src/sql/copy_table.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::pin::Pin;
+
+use common_query::physical_plan::SessionContext;
+use common_query::Output;
+use common_recordbatch::adapter::DfRecordBatchStreamAdapter;
+use datafusion::parquet::arrow::ArrowWriter;
+use datafusion::parquet::basic::{Compression, Encoding};
+use datafusion::parquet::file::properties::WriterProperties;
+use datafusion::physical_plan::RecordBatchStream;
+use futures::TryStreamExt;
+use object_store::services::fs::Builder;
+use object_store::ObjectStore;
+use snafu::ResultExt;
+use table::engine::TableReference;
+use table::requests::CopyTableRequest;
+
+use crate::error::{self, Result};
+use crate::sql::SqlHandler;
+
+impl SqlHandler {
+    pub(crate) async fn copy_table(&self, req: CopyTableRequest) -> Result<Output> {
+        let table_ref = TableReference {
+            catalog: &req.catalog_name,
+            schema: &req.schema_name,
+            table: &req.table_name,
+        };
+        let table = self.get_table(&table_ref)?;
+
+        let stream = table
+            .scan(None, &[], None)
+            .await
+            .with_context(|_| error::CopyTableSnafu {
+                table_name: table_ref.to_string(),
+            })?;
+
+        let stream = stream
+            .execute(0, SessionContext::default().task_ctx())
```

Comment:
The partition number is hardcoded; it looks like not correct. @v0y4g3r

Reply:
> The partition number is hardcoded; it looks like not correct. @v0y4g3r

The output partitions of query engine does not correspond to regions in storage engine, whatever value here is, `MitoTable::scan` will always go thru all existing regions in current datanode. 

https://github.com/GreptimeTeam/greptimedb/blob/374acc88303bb37616880a2077b943e76f050310/src/mito/src/table.rs#L149-L152

But just like the comment said, we should find a way to pass region parameters to `MitoTable`. Better add a todo here.

Reply:
Brother Lei told me to fill in 0 @v0y4g3r